test/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_string_buffer`
+
+class A {
+  String buffer;
+
+  void foo(int n) {
+    int aux = n;
+    while(aux --> 0) {
+      buffer += 'a'; // LINT
+    }
+  }
+}
```

Comment:
Yes it is linted, this rule in particular is not about what is easier to read, is about performance, in that way I think we should lint any += (or equivalents) inside a loop of a variable declared outside.

I know these cases are not easy to replace with a StringBuffer, but with the spirit of the rule (performance) they should be linted and detected and if the user is doing something like this and like the rule it should make an effort to improve performance with an StringBuffer or add a comment, that is my opinion.

What do you think is better?